updated logic for the client Order UI

AnsonYeong opened the pull request:
@AlphabetsAlphabets i have already update the clientOrder logic, haven't tested. Please help to test to see the logic can use or not

AnsonYeong committed:
updated logic for the client Order UI

AlphabetsAlphabets committed:
Merge remote-tracking branch 'origin/client-login' into client-login

AlphabetsAlphabets commented:
@AnsonYeong only got one problem.

![image](https://github.com/user-attachments/assets/23c2b853-2af3-48f2-9145-db483542ff1f)

I set Nasi Lemak quantity to 5. If I type in 10 and then click "OK". I order 5 Nasi Lemak.

![image](https://github.com/user-attachments/assets/9ea003db-48c9-4698-a13c-73b0c96a9529)

![image](https://github.com/user-attachments/assets/910787d7-17e6-438e-af22-de33ea9c0a75)

So, if I click on Nasi Lemak button. I can choose quantity using arrow. If I use arrow 3 times, quantity is 3. If I type 15 and then "OK". System says I will buy 3 Nasi Lemak.

AlphabetsAlphabets committed:
refactor: grouped imports

AlphabetsAlphabets committed:
fix: revert to minimum when amount over max

Solved a bug where if a user typed in the text field of the spinner an amount greater than what is allowed, they could order but not at the maximum they set but another value instead. This new addition warns them that they breached the max and reverts the value back to the minimum.
